composer.json

```diff
@@ -16,28 +16,23 @@
     ],
     "require": {
         "php": ">=7.1",
-        "symfony/console": "~3.0|~4.0|~5.0",
-        "symfony/filesystem": "~3.0|~4.0|~5.0",
-        "symfony/finder": "~3.0|~4.0|~5.0",
-        "symfony/process": "~3.0|~4.0|~5.0",
-        "symfony/yaml": "~3.0|~4.0|~5.0",
-        "twig/twig": "~1.10",
-        "erusev/parsedown": "~1.7.4",
-        "erusev/parsedown-extra": "~0.8",
-        "phine/phar": "~1.0",
-        "mnapoli/front-yaml": "~1.5",
-        "php-di/php-di": "^5.2.1",
-        "psr/log": "~1.0",
+        "symfony/console": "^3.0|^4.0|^5.0",
+        "symfony/filesystem": "^3.0|^4.0|^5.0",
+        "symfony/finder": "^3.0|^4.0|^5.0",
+        "symfony/process": "^3.0|^4.0|^5.0",
+        "symfony/yaml": "^3.0|^4.0|^5.0",
+        "twig/twig": "^1.10",
+        "erusev/parsedown": "^1.7",
+        "erusev/parsedown-extra": "^0.8",
+        "phine/phar": "^1.0",
+        "mnapoli/front-yaml": "^1.5",
+        "php-di/php-di": "^5.2",
+        "psr/log": "^1.0",
         "padraic/phar-updater": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~7.5",
+        "phpunit/phpunit": "^7.5|^8.5|^9.3",
         "squizlabs/php_codesniffer": "^3.3",
         "vimeo/psalm": "^3.16"
-    },
-    "config": {
-        "platform": {
-            "php": "7.1.3"
```

Comment:
Why locking the PHP version to 7.1.3, since all versions are tested in the CI?

Reply:
https://github.com/CouscousPHP/Couscous/pull/259#discussion_r491443886 ?

Reply:
This config is problematic because it tells Composer the PHP version installed is PHP 7.1.3 (see https://getcomposer.org/doc/06-config.md#platform).
It means that, in the CI, when PHP 7.4 is used, Composer is actually installing packages for PHP 7.1.3.

Reply:
I don't know the initial reason of why the platform was set to php 7.1, I only specified the bugfix version so I could install psalm.

If there isn't a concrete reason to specify the php version it's better to remove it indeed. That being said I don't think it's problematic that composer install dependencies targeting 7.1, as long as the dependencies themselves are tested against 7.4.

Reply:
I think it's problematic, because it means you cannot test Couscous against the latest dependencies.
For instance, Symfony 5 was not tested (PHP > 7.2), and, if tested, we could have seen it was failing because of the missing return codes in the commands.